tests/integ/modin/test_scikit.py

```diff
@@ -7,39 +7,19 @@
 from sklearn.preprocessing import MaxAbsScaler
 
 import snowflake.snowpark.modin.plugin  # noqa: F401
-from tests.integ.utils.sql_counter import SqlCounter
+from tests.integ.utils.sql_counter import sql_count_checker
 
 
-# SNOW-1344931 Support MaxAbsScalar on earlier versions of
-# scikit-learn by supporting numpy.may_share_memory
-def test_scikit_maxabs():
+@sql_count_checker(query_count=5)
+def test_maxabs():
     data = [[1.0, -1.0, 2.0], [2.0, 0.0, 0.0], [0.0, 1.0, -1.0]]
     X = pd.DataFrame(data)
-    # the following will result in a TypeError on earlier versions
-    # of scikit-learn if `numpy.may_share_memory' is not implemented as an
-    # array function. In later versions a NotImplementedError is
-    # thrown.
-    with SqlCounter(query_count=0):
-        try:
-            MaxAbsScaler().fit_transform(X)
-            raise AssertionError()
-        except NotImplementedError:
-            # scikit-learn 1.5.2 will throw an error for the DF
-            # interchange protocol.
-            pass
+    MaxAbsScaler().fit_transform(X)
 
 
-# SNOW-1518382 Support PCA  on earlier versions of
-# scikit-learn by supporting numpy.may_share_memory
-def test_scikit_pca():
+@sql_count_checker(query_count=3)
+def test_pca():
```

Comment:
Aren't there other 3rd party libraries that we know can now benefit from implementing the interchange protocol? If so, we should consider adding integration tests for them as well. Doesn't need to be in this PR though.

Reply:
There are more such third party libraries, and we also want to extend this scikit-learn test suite. We have follow-up tasks to add more tests.